src/stream.ts

```diff
@@ -53,7 +53,7 @@ export class DataAndMoneyStream extends Duplex {
   protected bytesRead: number
 
   constructor (opts: StreamOpts) {
-    super()
+    super({ allowHalfOpen: false })
     this.id = opts.id
```

Comment:
Added this since its the default we want to use.

Reply:
Will that change the behavior in any way here?

Reply:
I have not seen any behavior changes, but I noticed that when debugging it was set to true, feels like we should probably set it to false since thats what we are going for by default.